### ISSUE TITLE
RunPython timeout on Excel for Mac

### ISSUE THREAD

**Issue by goroderickgo** [xlwings/xlwings#2428]
#### OS (e.g. Windows 10 or macOS Sierra)

macOS Sonoma

#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)

- Python 3.11.7
- xlwings 0.29.0
- Excel Microsoft 365 16.84 (24041420)

#### Describe your issue (incl. Traceback!)

I have users that are encountering an "Apple even timeout" exception when using `RunPython` in a spreadsheet's VBA to call a Python function. The function works on smaller similar spreadsheets, but as this spreadsheet has grown, the function has to parse more ranges and seems to hit a hard-coded timeout limit.

I've seen [this slightly different issue & solution](https://github.com/xlwings/xlwings/issues/2297), which seems to describe calling a VBA macro from Python, but I'm not sure if there's an equivalent when calling Python from VBA. 

![image](https://github.com/xlwings/xlwings/assets/17296713/ef7b26f6-e821-4f53-bca8-b57ada5eb541)

#### Include a minimal code sample to reproduce the issue (and attach a sample workbook if required!)

[It will take me some time to mock up a minimal code example, but I think just creating a function that is called from `RunPython` that runs for more than 2 minutes will cause the timeout]

**Comment by goroderickgo** [comment]
I narrowed down a bit where it seems the AppleScript event is timing out when trying to get all names in the workbook (i.e., `wb.names`)... For reference, this workbook has ~150 named ranges. I haven't tested this consistently, but a few months ago, we weren't having timeout issues.

**Comment by goroderickgo** [comment]
I was able to run this workbook a few months ago, and it hasn't changed significantly since then. 

Another potentially related issue (sorry if this is veering off-course), is that it seems like converting large ranges (e.g., to pd.DataFrame) has gotten slower. For example, I have a 25 x 10,000 range, and it takes ~1 minute to convert to a DataFrame. Last time I used this workbook, this conversion happened almost instantaneously and I didn't run into any timeouts, but now I need to add `chunksize=1000` to avoid timeouts. 

I realize that it is hard to pinpoint why this might be happening, so appreciate just having the conversation and keeping tabs if any other users run into similar performance issues.
